src/lib/Libpython/pbs_python_svr_internal.c

```diff
@@ -2535,12 +2535,17 @@ pbs_python_populate_svrattrl_from_python_class(PyObject *py_instance,
 	PyObject	*py_resc = (PyObject *)NULL;
 	long		val_sec;
 	char		*objname = NULL;
+	char		*tmp = NULL;
 
 	if( the_val == NULL) {
-		the_val = (char *) malloc( HOOK_BUF_SIZE * sizeof(char));
+		the_val = (char *)malloc(HOOK_BUF_SIZE*sizeof(char));
 	}
 	if( the_resc == NULL) {
-		the_resc = (char *) malloc( HOOK_BUF_SIZE * sizeof(char));
+		the_resc = (char *)malloc(HOOK_BUF_SIZE*sizeof(char));
+	}
+
+	if( the_val == NULL || the_resc == NULL){
+		log_err(PBSE_SYSTEM, __func__, "Out of memory error");
```

Comment:
we need to return error on a failure instead of continuing further if any of the calls to malloc() fails.
Also, it is good to check after every malloc() call, as it makes sure that we do not call malloc() for the second time, if the first one failed.